cmd/ddev/cmd/root.go

```diff
@@ -119,6 +122,10 @@ Support: https://ddev.readthedocs.io/en/stable/users/support/`,
 // Execute adds all child commands to the root command sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
+	if _, _, err := RootCmd.Find(os.Args[1:]); err != nil {
+		suggestAddonForUnknownCommand(os.Args[1])
+	}
+
 	if err := RootCmd.Execute(); err != nil {
```

Comment:
That's because `RootCmd.Execute()` throws some error message that I can't prevent from being shown without intercepting it, which would make the code more complex.

```
$ ddev foobar
Error: unknown command "foobar" for "ddev"
Run 'ddev --help' for usage.
```

The above happens before the code goes inside the `if` you're asking about.